view/preview/helpers.js

```diff
@@ -0,0 +1,58 @@
+export const SVG_DOCTYPE = "http://www.w3.org/2000/svg";
+
+export function getScrollPercent(scrollHeight) {
+  const doc = document.documentElement;
+  const scrollTop = doc.scrollTop || document.body.scrollTop;
+
+  return (scrollTop / (scrollHeight - doc.clientHeight)) * 100;
+}
+
+export function getDimensions() {
+  const doc = document.documentElement;
+
+  return {
+    vw: Math.max(doc.clientWidth || 0, window.innerWidth || 0),
+    vh: Math.max(doc.clientHeight || 0, window.innerHeight || 0),
+  };
+}
+
+export function setSvgPosition({ svg, width, heigth }) {
+  svg.setAttribute("width", width);
+  svg.setAttribute("heigth", heigth);
+  svg.setAttribute("viewBox", `0 0 ${width} ${heigth}`);
+
+  return svg;
+}
+
+export function useScroll({ element, onStartScroll, onScroll, onEndScroll }) {
+  let isScroll = false;
+  let timer = null;
+
+  window.addEventListener(
+    "scroll",
+    () => {
+      let scrollPosition = getScrollPercent(element.offsetHeight);
+
+      if (isScroll) {
+        // onScroll
+        onScroll(scrollPosition);
+
+        // onEndScroll
+        if (timer !== null) {
+          clearTimeout(timer);
+        }
+        timer = setTimeout(function () {
+          isScroll = false;
+          onEndScroll(scrollPosition);
+        }, 150);
+      } else {
+        // onStartScroll
+        onStartScroll(scrollPosition);
+        isScroll = true;
```

Comment:
Do we need duplicate comments `onScroll`, `onEndScroll', `onEndScroll'? It seems that everything is quite concise as it is.

Reply:
done